plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2276,6 +2312,18 @@ void producer_plugin_impl::produce_block() {
 void producer_plugin::log_failed_transaction(const transaction_id_type& trx_id, const char* reason) const {
    fc_dlog(_trx_failed_trace_log, "[TRX_TRACE] Speculative execution is REJECTING tx: ${txid} : ${why}",
            ("txid", trx_id)("why", reason));
+
+   if (_trx_trace_dump_log.is_enabled(fc::log_level::debug)){
+       transaction_metadata_ptr trx_meta_ptr = my->_unapplied_transactions.get_trx(trx_id);
```

Comment:
It will never be in the `unapplied_transactions`. Modify `log_failed_transaction` to take a `const packed_transaction_ptr&`

Reply:
When is a transaction put into the `unapplied_transaction_queue`?

Reply:
See calls to `unapplied_transaction_queue` `add_forked`, `add_aborted`, `add_persisted`

Reply:
I did search `unapplied_transactions` in the code to see where a trx could be in the queue, but was not sure. Thanks!

Reply:
Fixed.